### ISSUE TITLE
Make slickNext, slickPrev and slickGoTo return index of the current slide

### ISSUE THREAD

**Issue by lukaske** [akiran/react-slick#1878]
_tl;dr: Functions slickNext, slickPrev and slickGoTo don't return anything. It would be great if they returned a promise and/or the index of the currently active slide_

Hi!

This library is awesome, but I figured it is missing some features. I was trying to implement a custom slider dots indicator component, but I was having a hard time syncing the dots with react-slick. 

Functions slickNext, slickPrev and slickGoTo don't return anything. If they returned a promise I would be able to keep track of slides inside a wrapper component, currently it just gets out of sync. It would be even better if they returned an index I could use that to sync the dots directly. Currently neither of these features are supported.

Edit: beforeChange and afterChange exist for this purpose!
